engine/utils/HeapArray.h

```diff
@@ -0,0 +1,735 @@
+//
+//  Copyright (c) 2022 Jonathan Moallem (@J-Mo63) & Aryeh Zinn (@Raelr)
+//
+//  This code is released under an unmodified zlib license.
+//  For conditions of distribution and use, please see:
+//      https://opensource.org/licenses/Zlib
+//
+
+#ifndef SIEGE_ENGINE_HEAP_ARRAY_H
+#define SIEGE_ENGINE_HEAP_ARRAY_H
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <initializer_list>
+#include <utility>
+
+typedef const size_t sizet;
```

Comment:
This is quite unnecessary

Reply:
you are correct - I remember this was leftover from a small experiment.